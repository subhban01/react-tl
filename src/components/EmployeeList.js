import React from 'react';
import BEMHelper from 'react-bem-helper';

let employeeList = new BEMHelper('employee-list');
class EmployeeList extends React.Component {

    constructor(props) {
        super(props);
        this.onCardClick = this.onCardClick.bind(this);
        this.state = {
            activeItem: -1
        }
    }

    onCardClick(event, index) {
        this.setState({
            activeItem: index
        })
        this.props.showModalPopUp(index);
    }

    componentWillReceiveProps(nextProps) {
        if(!nextProps.modalOpen) {
            this.setState({
                activeItem: -1
            })
        }
    }

    render(){
        const {employees} = this.props;
        const {activeItem} = this.state;
        const employeeItems = employees.map((employee, index) => {
            return (
                <article {...employeeList('card', '', activeItem === index ?'selected-item' : '')} key={index}>
                    <a {...employeeList('card-anchor')} onClick={(event) => this.onCardClick(event, index)}>
                        <figure {...employeeList('thumbnail')}>
                            <img {...employeeList('thumbnail-img')} src={employee.avatar}/>
                        </figure>
                        <div {...employeeList('card-content')}>
                            <div {...employeeList('card-content', '', 'content-header')}>
                                <span {...employeeList('card-content', '', 'first-name')}>{employee.firstName}</span>
                                <span {...employeeList('card-content', '', 'last-name')}>{employee.lastName}</span>
                            </div>
                            <p {...employeeList('card-content', '', 'paragraph')}>{employee.bio}</p>
                        </div>
                    </a>
                </article>
            )
        });

        return (
            <div {...employeeList('main-area')}>
                <div {...employeeList('main-area', 'centered')}>
                    <section {...employeeList('cards')}>
                        {employeeItems}
                    </section>
                </div>
            </div>
        );
    }
}

export default EmployeeList;