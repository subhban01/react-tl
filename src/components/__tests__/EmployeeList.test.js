import * as React from 'react';
import {shallow} from 'enzyme';
import EmployeeList from '../EmployeeList.js';


describe('EmployeeList Component', () => {
    it('should render the employeeList correctly', () => {
        const employees = [
            {
                "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/joannefournier/128.jpg",
                "firstName": "Ali",
                "lastName": "Kiehn",
                "jobTitle": "Dynamic Configuration Representative",
                "age": 17,
                "bio": "Dolorem voluptatibus rem repellat qui repellat fuga quod veniam itaque. Ad fugit repudiandae ut voluptas aut. Iusto fuga eos accusamus et unde alias. Totam et veritatis qui et minus.",
                "dateJoined": "2012-02-13T15:56:38.391Z"
            },
            {
                "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/alek_djuric/128.jpg",
                "firstName": "Tahlia",
                "lastName": "Rees",
                "jobTitle": "Lead Directives Designer",
                "age": 44,
                "bio": "Quasi iste aspernatur nam voluptatem doloribus. Assumenda molestiae ut ut natus assumenda nemo error. Ab accusamus sunt enim. In possimus dolor quo. Maiores natus voluptatem exercitationem delectus. Error tempore non quo voluptas.",
                "dateJoined": "2017-08-26T12:29:45.996Z"
            },
            {
                "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/joetruesdell/128.jpg",
                "firstName": "Isabella",
                "lastName": "D'amore",
                "jobTitle": "Regional Operations Coordinator",
                "age": 50,
                "bio": "Odit possimus dolores mollitia aperiam labore. Qui tempore culpa illo iusto beatae a voluptatem atque rerum. Aliquid atque quibusdam et qui dolorem ut voluptatem.",
                "dateJoined": "2014-02-26T16:05:16.483Z"
            },
        ]

        const wrapper = shallow(
            <EmployeeList employees={employees}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
})
;