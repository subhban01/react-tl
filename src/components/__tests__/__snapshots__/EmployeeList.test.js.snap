// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeList Component should render the employeeList correctly 1`] = `
<div
  className="employee-list__main-area"
>
  <div
    className="employee-list__main-area employee-list__main-area--centered"
  >
    <section
      className="employee-list__cards"
    >
      <article
        className="employee-list__card"
        key="0"
      >
        <a
          className="employee-list__card-anchor"
          onClick={[Function]}
        >
          <figure
            className="employee-list__thumbnail"
          >
            <img
              className="employee-list__thumbnail-img"
              src="https://s3.amazonaws.com/uifaces/faces/twitter/joannefournier/128.jpg"
            />
          </figure>
          <div
            className="employee-list__card-content"
          >
            <div
              className="employee-list__card-content content-header"
            >
              <span
                className="employee-list__card-content first-name"
              >
                Ali
              </span>
              <span
                className="employee-list__card-content last-name"
              >
                Kiehn
              </span>
            </div>
            <p
              className="employee-list__card-content paragraph"
            >
              Dolorem voluptatibus rem repellat qui repellat fuga quod veniam itaque. Ad fugit repudiandae ut voluptas aut. Iusto fuga eos accusamus et unde alias. Totam et veritatis qui et minus.
            </p>
          </div>
        </a>
      </article>
      <article
        className="employee-list__card"
        key="1"
      >
        <a
          className="employee-list__card-anchor"
          onClick={[Function]}
        >
          <figure
            className="employee-list__thumbnail"
          >
            <img
              className="employee-list__thumbnail-img"
              src="https://s3.amazonaws.com/uifaces/faces/twitter/alek_djuric/128.jpg"
            />
          </figure>
          <div
            className="employee-list__card-content"
          >
            <div
              className="employee-list__card-content content-header"
            >
              <span
                className="employee-list__card-content first-name"
              >
                Tahlia
              </span>
              <span
                className="employee-list__card-content last-name"
              >
                Rees
              </span>
            </div>
            <p
              className="employee-list__card-content paragraph"
            >
              Quasi iste aspernatur nam voluptatem doloribus. Assumenda molestiae ut ut natus assumenda nemo error. Ab accusamus sunt enim. In possimus dolor quo. Maiores natus voluptatem exercitationem delectus. Error tempore non quo voluptas.
            </p>
          </div>
        </a>
      </article>
      <article
        className="employee-list__card"
        key="2"
      >
        <a
          className="employee-list__card-anchor"
          onClick={[Function]}
        >
          <figure
            className="employee-list__thumbnail"
          >
            <img
              className="employee-list__thumbnail-img"
              src="https://s3.amazonaws.com/uifaces/faces/twitter/joetruesdell/128.jpg"
            />
          </figure>
          <div
            className="employee-list__card-content"
          >
            <div
              className="employee-list__card-content content-header"
            >
              <span
                className="employee-list__card-content first-name"
              >
                Isabella
              </span>
              <span
                className="employee-list__card-content last-name"
              >
                D'amore
              </span>
            </div>
            <p
              className="employee-list__card-content paragraph"
            >
              Odit possimus dolores mollitia aperiam labore. Qui tempore culpa illo iusto beatae a voluptatem atque rerum. Aliquid atque quibusdam et qui dolorem ut voluptatem.
            </p>
          </div>
        </a>
      </article>
    </section>
  </div>
</div>
`;
